src/hotspot/os/posix/signals_posix.cpp

```diff
@@ -141,12 +146,12 @@ class SavedSignalHandlers {
 debug_only(static bool signal_sets_initialized = false);
 static sigset_t unblocked_sigs, vm_sigs, preinstalled_sigs;
 
-// For CheckJNI:
-//  Our own hotspot signal handlers should never ever get replaced by a third
-//  party one. To check that, store a copy of the handler setup and compare it
-//  periodically against reality (see os::run_periodic_checks()).
+// Our own signal handlers should never ever get replaced by a third party one.
+//  To check that, and to aid with diagnostics, store a copy of the handler setup
+//  and compare it periodically against reality (see os::run_periodic_checks()).
 static bool check_signals = true;
 static SavedSignalHandlers expected_handlers;
+static bool do_check_signal_periodically[NSIG];
```

Comment:
Does this need explicit initialization to have false entries?

Reply:
I think it does not since global static objects are zero-initialized and 0 = false; however, I made the initialization explicit.